### PR TITLE
dev-lang/nim: more improvements

### PR DESCRIPTION
**Pull request by CyberTailor**
* build with linenoise unconditionally
* fix "calls cc directly" in stage1

### CODE REVIEW

File: dev-lang/nim/nim-1.6.6-r2.ebuild [hunk 1]
```diff
@@ -65,6 +59,8 @@ src_prepare() {
 	# note: there are consumers in the ::guru overlay
 	use experimental && \
 		eapply "${WORKDIR}"/${PN}-patches-${PV}
+
+	sed "s/-O3//" -i build.sh || die
```
Comment (xgqt) [inline dev-lang/nim/nim-1.6.6-r2.ebuild:63]: patch plz

also, are you sure we do not have to replace it with user CFLAGS?

File: dev-lang/nim/nim-1.6.6-r2.ebuild [hunk 2]
```diff
@@ -15,11 +15,10 @@ SRC_URI="
 LICENSE="MIT"
 SLOT="0"
 KEYWORDS="~amd64 ~arm ~x86"
-IUSE="debug experimental +readline"
+IUSE="debug experimental"
 RESTRICT="test"  # need to sort out depends and numerous failures
 
-RDEPEND="readline? ( sys-libs/readline:0= )"
-DEPEND="${RDEPEND}"
+BDEPEND="<sys-process/parallel-20220622"  # bug 856079
```
Comment (xgqt) [inline dev-lang/nim/nim-1.6.6-r2.ebuild:21]: I don't get it.
This one: https://bugs.gentoo.org/856079 ?

Comment (thesamesam) [reply dev-lang/nim/nim-1.6.6-r2.ebuild:21]: It's a bug which needs fixing but this is really not the right fix. You could argue this for every single reverse dep of parallel.

Comment (CyberTailor) [reply dev-lang/nim/nim-1.6.6-r2.ebuild:21]: Yes

Comment (thesamesam) [reply dev-lang/nim/nim-1.6.6-r2.ebuild:21]: Fixed properly now. In future, just ping. Please no hacks like this.

File: dev-lang/nim/nim-1.6.6-r2.ebuild [hunk 3]
```diff
@@ -15,25 +15,22 @@ SRC_URI="
...
 # BDEPEND="test? ( net-libs/nodejs )"
 
-PATCHES=( "${FILESDIR}"/${PN}-0.20.0-paths.patch )
+PATCHES=(
+	"${FILESDIR}"/${PN}-0.20.0-paths.patch
+	"${FILESDIR}"/${PN}-1.6.6-csources-flags.patch
+)
 
 _run() {
 	einfo "Running: ${@}"
```
Comment (thesamesam) [inline dev-lang/nim/nim-1.6.6-r2.ebuild:30]: edo?

File: dev-lang/nim/nim-1.6.6-r2.ebuild (hunk 2 above)
Comment (CyberTailor) [reply dev-lang/nim/nim-1.6.6-r2.ebuild]: Removed the hack

Comment (thesamesam) [reply dev-lang/nim/nim-1.6.6-r2.ebuild]: thanks!

File: dev-lang/nim/nim-1.6.6-r2.ebuild (hunk 3 above)
Comment (CyberTailor) [reply dev-lang/nim/nim-1.6.6-r2.ebuild:30]: no, it also sets PATH

Comment (thesamesam) [reply dev-lang/nim/nim-1.6.6-r2.ebuild:30]: ... just do PATH=... edo?

Comment (xgqt) [reply dev-lang/nim/nim-1.6.6-r2.ebuild:30]: 4x _run == 4x PATH=... edo ...
kinda ugly imo

maybe 
```bash
_run() {
    PATH=... edo ...
}
```

**Comment by CyberTailor** [comment]
Only with huge patches

**Comment by thesamesam** [comment]
It might work to try do parallel --citation or something and set the homedir to ${T}. Dunno.

**Comment by xgqt** [comment]
@CyberTailor failed build with parallel:
[dev-lang:nim-1.6.6-r2:20220714-112143.log](https://github.com/gentoo/gentoo/files/9111424/dev-lang.nim-1.6.6-r2.20220714-112143.log)

**Comment by xgqt** [comment]
Thanks a lot!

**Comment by CyberTailor** [comment]
why squash...

**Pull request closed without merging**